### DISCUSSION TITLE
Allow variable width flow tabs (UI feature request)

### DISCUSSION THREAD

**Discussion by jpravetz** [node-red/designs#83] [Feature Requests]
This is a _nice to have_ feature request.

If you have a lot of flows/subflows the number of tabs open will exceed the space and you must scroll to see the hidden tabs. Very conveniently, a mouse scroll wheel can be used for this. 

However, it would be nice if the tabs were more like in a browser (eg. Firefox), where they squish to a narrower width, so that they all fit, up to the point where the tabs are too narrow (just like browsers implement). Alternatively you can do like in VSCode and make variable width tabs, dependent on their titles. My opinion is that the browser-style implementation, where all tabs are the same width, is better. But either solution would be an improvement.

These are the tabs I am referring too (in case there is confusion):
![Screenshot 2023-09-13 at 1 34 34 PM](https://github.com/node-red/designs/assets/154432/2e964b73-4d8c-406d-abec-72f9c70e772d)

**Comment by knolleary** [comment]
Hi @jpravetz 

Thanks for the request.

Over the history of the project we've tried a few different approaches for the tabs, and had plenty of feedback from the community with each iteration.

We did at one point do as you describe; shrink the tabs, but we found you needed a reasonable minimum size otherwise it became hard to distinguish - at which point, the benefit of shrinking was a bit lost. Unlike browsers, we don't have favicons to help identify the different tabs when they are shrunk down.

I'll also note each browser has a slightly different behaviour. Chrome appears to just squash everything into the visible tab bar. Firefox shrinks but then overflows to making it scrollable.

So, whilst I won't claim what we have is perfect, it does come as the result of some amount of iteration.

**Comment by jpravetz** [comment]
I totally get your points. You've definitely been listening to users longer than I've been a user :-). 

In reality your `info panel` does substantially mitigate this issue. Although I'd probably prefer this panel be shared with the `node panel` in the left sidebar (since I have `debug` open the most, and since the width of `info` and `node` panels could both be made narrow like the `node panel` is now, whereas I make the `debug` and `help` panels wider).

Thanks for the response. Node-RED is a really nice tool with already great UI.
